new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java

```diff
@@ -257,6 +259,46 @@ public static V1DeploymentList listDeployments(String namespace) {
   }
 
   // --------------------------- pods -----------------------------------------
+  
+  /**
+   * Returns the V1Pod object given the following parameters.
+   * @param namespace in which to check for the pod existence
+   * @param labelSelector in the format "weblogic.domainUID in (%s)"
+   * @param podName name of the pod to return
+   * @return V1Pod object if found otherwise null
+   * @throws ApiException when there is error in querying the cluster
+   */
+  public static V1Pod getPod(String namespace, String labelSelector, String podName) throws ApiException {
+    V1PodList pods = listPods(namespace, labelSelector);
+    for (var pod : pods.getItems()) {
+      if (podName.equals(pod.getMetadata().getName())) {
+        return pod;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Get the creationTimestamp for a given pod with following parameters.
+   * @param namespace in which to check for the pod existence
+   * @param labelSelector in the format "weblogic.domainUID in (%s)"
```

Comment:
labelSelector in the format "weblogic.domainUID in (%s)", can be null

Reply:
yes, labelSelector can be a null String